rllib/core/rl_module/rl_module.py

```diff
@@ -59,13 +59,18 @@ class SingleAgentRLModuleSpec:
             observation space of an environment, would usually correspond to a
             one-hot encoded observation space of the RLModule because of preprocessing.
         action_space: The action space of the RLModule.
+        algorithm_config_overrides: An optional AlgorithmConfig override dict that
```

Comment:
I don't think we should retrofit SingleAgentRLModuleSpecs to take bits of information that RLModule won't use directly. It's just not the right place. algorithm_config_overrides does not belong to SingleAgentRLModuleSpec
We only use this metric when creating a LearnerHyperParameters's override dict, right? So we can put this information in algorithm_config directly. For example we could do the following to pass in the algorithm_config_overrides.

```
config = config.multi_agent(algorithm_config_overrides)
```

Reply:
I was thinking about this, too. I think you are right. This information would sit at the wrong location.
It's actually the Learner that should hold this information on which Module receives which config override, not the module itself. I'll move this into the higher algo-config level directly, then derive the per-module settings within the Learner c'tor and keep things outside of Single|MultiAgentRLModuleSpec.

Reply:
done